javascript/net/grpc/web/generator/build.zig

```diff
@@ -0,0 +1,215 @@
+const std = @import("std");
+const CrossTarget = std.zig.CrossTarget;
+
+fn format(comptime fmt: []const u8, args: anytype) []const u8 {
+    return std.fmt.allocPrint(std.testing.allocator, fmt, args) catch unreachable;
+}
+
+const BinaryTarget = struct {
+    name: []const u8,
+    arch: []const u8,
+};
+
+pub fn build(b: *std.build.Builder) void {
+    // Standard release options allow the person running `zig build` to select
+    // between Debug, ReleaseSafe, ReleaseFast, and ReleaseSmall.
+    const mode = b.standardReleaseOptions();
+
+    var version = if (std.os.getenv("VERSION")) |v| v else "unknown";
+
+    var targets = [_]BinaryTarget{
+        // for now, let's only build aarch64 binaries
```

Comment:
The most significant difference is that in the existing workflow we build x86_64 using GCC, and Zig builds binaries using Clang internally (you can imagine Zig as a Clang wrapper). I think both of them are well optimized since Clang is also a very popular compiler. There are some binary size differences, but personally I'm OK with them.

Binary sizes with Zig:

```
protoc-gen-grpc-web-unknown-darwin-x86_64               6.7M
protoc-gen-grpc-web-unknown-linux-x86_64                1.8M
protoc-gen-grpc-web-unknown-windows-x86_64.exe          4.3M
```

Binary sizes with current workflow:

<img width="915" alt="CleanShot 2022-06-03 at 08 55 56@2x" src="https://user-images.githubusercontent.com/15059605/171766423-6fdb2aed-10dd-41fb-a4f5-c90411b0af1d.png">

Reply:
ohh that's very cool to know that Zig is Clang based! Thanks a lot for the explanation! The binary sizes do seem to be around the same ballpark too so that's great to know too! :)

Maybe we'll consider moving all to Zig later if Zig turns out be be working well and stable :)